table/block_based/block_based_table_reader.cc

```diff
@@ -3202,6 +3204,7 @@ InternalIterator* BlockBasedTable::NewIterator(
     size_t compaction_readahead_size) {
   BlockCacheLookupContext lookup_context{caller};
   bool need_upper_bound_check =
+      read_options.auto_prefix_mode ||
```

Comment:
When auto_prefix_mode=true and PrefixExtractorChanged=false ,user set rep_->index_type == BlockBasedTableOptions::kHashSearch, then user will not to able to use prefix seek. It seems to me that auto_prefix_mode and kHashSearch can't be used at the same time, Can you help me explain why?